mycroft/skills/skill_loader.py

```diff
@@ -263,95 +273,63 @@ def _prepare_for_load(self):
         self.instance = None
 
     def _skip_load(self):
-        log_msg = 'Skill {} is blacklisted - it will not be loaded'
-        LOG.info(log_msg.format(self.skill_id))
+        LOG.info(f'Skill {self.skill_id} is blacklisted - it will not be loaded')
 
     def _load_skill_source(self):
         """Use Python's import library to load a skill's source code."""
         main_file_path = os.path.join(self.skill_directory, SKILL_MAIN_MODULE)
+        skill_module = None
         if not os.path.exists(main_file_path):
-            error_msg = 'Failed to load {} due to a missing file.'
-            LOG.error(error_msg.format(self.skill_id))
+            LOG.error(f'Failed to load {self.skill_id} due to a missing file.')
         else:
             try:
                 skill_module = load_skill_module(main_file_path, self.skill_id)
             except Exception as e:
-                LOG.exception('Failed to load skill: '
-                              '{} ({})'.format(self.skill_id, repr(e)))
-            else:
-                module_is_skill = (
-                    hasattr(skill_module, 'create_skill') and
-                    callable(skill_module.create_skill)
-                )
-                if module_is_skill:
-                    return skill_module
-        return None  # Module wasn't loaded
+                LOG.exception(f'Failed to load skill: {self.skill_id} ({e})')
+        return skill_module
 
     def _create_skill_instance(self, skill_module):
-        """Use v2 skills framework to create the skill."""
+        """create the skill object.
+
+        Arguments:
+            skill_module (module): Module to load from
+
+        Returns:
+            (bool): True if skill was loaded successfully.
+        """
         try:
-            self.instance = skill_module.create_skill()
+            skill_class = get_skill_class(skill_module)
+
+            # filter kwargs depending on what skill developers exposed
+            kwargs = {"skill_id": self.skill_id, "bus": self.bus}
+            kwargs = {k: v for k, v in kwargs.items()
+                      if k in signature(skill_class).parameters}
+
+            # create the skill
+            self.instance = skill_class(**kwargs)
+
+            if not self.instance.is_fully_initialized:
+                # finish initialization of skill class
+                # NOTE: sometimes this isn't called in MycroftSkill __init__ method
+                # because skills usually do not expose the required kwargs
```

Comment:
Should this be logged as a warning or something?

Like `please extend your __init__ to include and forward {missing_args}`

Reply:
not sure about this one, this is fully optional and we can't enforce a standard on mycroft skills. if mycroft-core ignores this we shouldnt scream at the users about it

Reply:
Sounds reasonable

Reply:
Right, that's quite a valid point.